sdks/python/apache_beam/runners/runner.py

```diff
@@ -187,7 +187,15 @@ def visit_transform(self, transform_node):
           logging.error('Error while visiting %s', transform_node.full_label)
           raise
 
+    # Replace the element type for inputs to a GroupByKey operation with a
+    # KV type hint that would allow runners to choose a standard coder
+    # (TupleCoder) for these elements.
     pipeline.visit(group_by_key_input_visitor())
+
+    # Replace the element type of inputs to a Flatten operation with that of the
+    # output of Flatten.
+    # TODO: This is only needed for the Dataflow runner, fix once the new runner
```

Comment:
The runner API won't change anything here. Please move it (and likely the GBK one). to DataflowRunner.run() now rather than leaving a lingering TODO.